metricbeat/mb/registry.go

```diff
@@ -144,9 +147,42 @@ func (r *Register) metricSetFactory(module, name string) (MetricSetFactory, Host
 	return info.factory, info.hostParser, nil
 }
 
+//Modules returns the list of module names that are registered
+func (r *Register) Modules() []string {
+	r.RLock()
+	defer r.RUnlock()
+
+	modules := []string{}
```

Comment:
Change this to `make([]string, 0, len(r.modules))` since it knows the capacity requirement a priori.